uaclient/cli.py

```diff
@@ -372,7 +385,24 @@ def _get_contract_token_from_cloud_identity(cfg: config.UAConfig) -> str:
 
     :return: contract token obtained from identity doc
     """
-    instance = identity.cloud_instance_factory()
+    try:
+        instance = identity.cloud_instance_factory()
+    except exceptions.UserFacingError as e:
+        if cfg.is_attached:
+            # We are attached on non-Pro Image, just report already attached
+            raise exceptions.AlreadyAttachedError(cfg)
+        # Unattached on non-Pro return UserFacing error msg details
+        raise e
+    current_iid = identity.get_instance_id()
+    if cfg.is_attached:
+        prev_iid = cfg.read_cache("instance-id")
+        if current_iid == prev_iid:
+            raise exceptions.AlreadyAttachedError(cfg)
+        print("Re-attaching Ubuntu Advantage subscription on new instance")
+        if _detach(cfg, assume_yes=True) == 1:
```

Comment:
Given this is an exit code, we can future-proof ourselves fully with:

```suggestion
        if _detach(cfg, assume_yes=True) != 0:
```

Reply:
done.